simplic-configurationanalyser/Simplic.ConfigurationAnalyzer.Service/Linter/PythonImportAnalyzer.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Dapper;
+using Simplic.Framework.Core;
+using Simplic.Sql;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Simplic.ConfigurationAnalyzer.Service
+{
+    /// <summary>
+    /// Python import linter
+    /// </summary>
+    public class PythonImportAnalyzer : IConfigurationTypeAnalyzerService
+    {
+        private ISqlService sqlService;
+
+        /// <summary>
+        /// Initialize analyzer
+        /// </summary>
+        /// <param name="sqlService">Sql service</param>
+        public PythonImportAnalyzer(ISqlService sqlService)
+        {
+            this.sqlService = sqlService;
+        }
+
+        /// <summary>
+        /// Analyze python imports
+        /// </summary>
+        /// <returns></returns>
+        public IList<Result> Analyze()
+        {
+            var results = new List<Result>();
+
+            sqlService.OpenConnection((connection) =>
+            {
+                var scripts = connection.Query<PythonScript>("select ScriptName as Name, ScriptContent as Code from ESS_MS_Dynamic_Script");
+                foreach (var script in scripts)
+                {
+                    foreach (var line in script.Code.Split(new char[] { '\r', '\n' }))
+                    {
+                        if (line.StartsWith("import ") || line.StartsWith("from "))
+                        {
+                            try
+                            {
+                                GlobalDlrHost.Host.DefaultScope.Execute(line);

```

Comment:
do you really want to execute the python script ? there should be an isolated python scope so that the script wont affect anything. you can check for syntax error but executing it directly could be harmfull or could have unwanted results: sending an email or sms, sending telematic messages, changing records in the database etc.

Reply:
Only import lines will be executed, no code that executes stuff. Doing it in a custom scope is a good idea.